Sources/Basic/FileSystem.swift

```diff
@@ -467,5 +466,60 @@ public class InMemoryFileSystem: FileSystem {
     }
 }
 
+/// A rerooted view on an existing FileSystem.
+///
+/// This is a simple wrapper which creates a new FileSystem view into a subtree
+/// of an existing filesystem. This is useful for passing to clients which only
+/// need access to a subtree of the filesystem but should otherwise remain
+/// oblivious to its concrete location.
+public struct RerootedFileSystemView: FileSystem {
+    /// The underlying file system.
+    private var underlyingFileSystem: FileSystem
+
+    /// The root path within the containing file system.
+    private let root: AbsolutePath
+    
+    public init(_ underlyingFileSystem: inout FileSystem, rootedAt root: AbsolutePath) {
```

Comment:
... this means that a client which got a FileSystem, but wants to expose an (readonly) `FileSystem` to a client, will still have to make a copy to create the rerooted FileSystem (even though the copy won't be modified). That's unfortunate, but at least is localized.